applications/GeoMechanicsApplication/custom_workflows/dgeoflow.cpp

```diff
@@ -597,24 +638,17 @@ namespace Kratos
         }
 
         // Get Find boundary in Processes
-        for (const auto& process : rProcesses)
+        for (const auto& process : mProcesses)
         {
-            ModelPart *currentModelPart;
+            ModelPart *currentModelPart = nullptr;
 
             if (process->Info() == "ApplyConstantScalarValueProcess")
             {
                 auto current_process = std::static_pointer_cast<GeoFlowApplyConstantScalarValueProcess>(process);
                 if (current_process->hasWaterPressure())
                 {
-                    currentModelPart = &current_process->GetModelPart();
-                    try
-                    {
-                        currentModelPart->GetNode(RiverNode);
-                        RiverBoundary = current_process;
-                    }
-                    catch (...)
-                    {
-                    }
+                    currentModelPart->GetNode(RiverNode);
```

Comment:
Here we miss the initialization of currentModelPart, so it will be a nullptr when calling it. We still need the `currentModelPart = &current_process->GetModelPart();` which was there before

Reply:
By the way, @indigocoral did this not result in one or more failing tests? Then we might not have a test related to this specific part